pkg/pillar/cmd/volumemgr/initialvolumestatus.go

```diff
@@ -113,19 +114,24 @@ func gcObjects(ctx *volumemgrContext, dirName string) {
 		return
 	}
 	for _, location := range locations {
+		filelocation := path.Join(dirName, location.Name())
 		volumeName := strings.Split(location.Name(), ".")
 		if len(volumeName) > 1 {
 			volumeKey := volumeName[0]
+			format := strings.ToUpper(volumeName[1])
 			vs := lookupVolumeStatus(ctx, volumeKey)
 			if vs == nil {
-				log.Errorf("gcObjects: Found unused volume %s in %s. Deleting it.",
-					location.Name(), dirName)
-				deleteFile(path.Join(dirName, location.Name()))
+				log.Infof("gcObjects: Found unused volume %s in %s. Deleting it.",
+					filelocation, dirName)
+				if zconfig.Format(zconfig.Format_value[format]) == zconfig.Format_CONTAINER {
+					_ = containerd.SnapshotRm(filelocation, true)
+				}
+				deleteFile(filelocation)
```

Comment:
Doesn't this still need a RemoveAll? Even if the SnapshotRm removes all of the content it is still a directory.

Reply:
We are already calling ```RemoveAll``` in ```deleteFile``` function.
```
func deleteFile(filelocation string) {
        if err := os.RemoveAll(filelocation); err != nil {
                log.Errorf("Failed to delete file %s. Error: %s",
                        filelocation, err.Error())
        }
}
```

Reply:
Then I don't understand why it failed. Perhaps I didn't read the error log carefully enough?

Reply:
Even I was also confused with the error message. In case the unmount needed before delete, we get resource busy error which was not there in the logs.